packages/wonder-blocks-dropdown/components/dropdown.js

```diff
@@ -16,33 +16,60 @@ import typeof ActionItem from "./action-item.js";
 import typeof OptionItem from "./option-item.js";
 import typeof SeparatorItem from "./separator-item.js";
 
+const keyCodes = {
+    tab: 9,
+    escape: 27,
+    space: 32,
+    up: 38,
+    down: 40,
+};
+
 type DropdownProps = {|
     /**
      * Items for the menu.
      */
     children: Array<React.Element<ActionItem | OptionItem | SeparatorItem>>,
 
     /**
-     * Whether the menu is open or not.
+     * An index that represents the index of the focused element when the menu
+     * is opened.
      */
-    open: boolean,
+    initialFocusedIndex: number,
 
     /**
-     * The component that opens the menu.
+     * A list of React component refs to the non-disabled items of this menu.
+     * The items should have a focusItem method, which is called by keyboard
+     * listeners in this component to provide keyboard accessible navigation.
      */
-    opener: React.Element<*>,
+    itemRefs: Array<{current: any}>,
```

Comment:
This is being passed by the wrapper components around Dropdown. Two reasons it's being passed instead of generated by the Dropdown component itself: I didn't know if it was possible to render all the children here adding a ref via `React.createRef` without having to reclone all the elements, and the wrapper components are filtering out disabled items and separator components before passing this, which is a good thing.